themes/default/content/docs/concepts/_index.md

```diff
@@ -15,262 +14,62 @@ aliases:
 
-{{% /choosable %}}
-{{% choosable language go %}}
-
-```go
-package main
-
-import (
-	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws/ec2"
-	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
-)
-
-func main() {
-	pulumi.Run(func(ctx *pulumi.Context) error {
-		group, err := ec2.NewSecurityGroup(ctx, "web-sg", &ec2.SecurityGroupArgs{
-			Description: pulumi.String("Enable HTTP access"),
-			Ingress: ec2.SecurityGroupIngressArray{
-				ec2.SecurityGroupIngressArgs{
-					Protocol:   pulumi.String("tcp"),
-					FromPort:   pulumi.Int(80),
-					ToPort:     pulumi.Int(80),
-					CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
-				},
-			},
-		})
-		if err != nil {
-			return err
-		}
-		server, err := ec2.NewInstance(ctx, "web-server", &ec2.InstanceArgs{
-			Ami:                 pulumi.String("ami-6869aa05"),
-			InstanceType:        pulumi.String("t2.micro"),
-			VpcSecurityGroupIds: pulumi.StringArray{group.Name},
-		})
-		if err != nil {
-			return err
-        }
-
-		ctx.Export("publicIp", server.PublicIp)
-		ctx.Export("publicHostName", server.PublicDns)
-		return nil
-	})
-}
-```
-
-{{% /choosable %}}
-{{% choosable language csharp %}}
-
-```csharp
-using System.Threading.Tasks;
-using Pulumi;
-using Pulumi.Aws.Ec2;
-using Pulumi.Aws.Ec2.Inputs;
-
-class Program
-{
-    static Task<int> Main() => Deployment.RunAsync<MyStack>();
-}
-
-class MyStack : Stack
-{
-    public MyStack()
-    {
-        var group = new SecurityGroup("web-sg", new SecurityGroupArgs {
-            Description = "Enable HTTP access",
-            Ingress = {
-                new SecurityGroupIngressArgs {
-                    Protocol = "tcp",
-                    FromPort = 80,
-                    ToPort = 80,
-                    CidrBlocks = { "0.0.0.0/0" }
-                }
-            }
-        });
-        var server = new Instance("web-server", new InstanceArgs {
-            Ami = "ami-6869aa05",
-            InstanceType = "t2.micro",
-            VpcSecurityGroupIds = { group.Name }
-        });
-        this.PublicIp = server.PublicIp;
-        this.PublicDns = server.PublicDns;
-    }
-
-    [Output]
-    public Output<string> PublicIp { get; set; }
-
-    [Output]
-    public Output<string> PublicDns { get; set; }
-}
-```
-
-{{% /choosable %}}
-{{% choosable language java %}}
-
-```java
-package myproject;
-
-import com.pulumi.Context;
-import com.pulumi.Exports;
-import com.pulumi.Pulumi;
-import com.pulumi.aws.ec2.Instance;
-import com.pulumi.aws.ec2.InstanceArgs;
-import com.pulumi.aws.ec2.SecurityGroup;
-import com.pulumi.aws.ec2.SecurityGroupArgs;
-import com.pulumi.aws.ec2.inputs.SecurityGroupIngressArgs;
+This diagram illustrates the structure and major components of Pulumi.
 
-import java.util.List;
+![Pulumi programming model diagram.](/images/docs/pulumi-programming-model-diagram.svg)
```

Comment:
It would be nice to have a graphic designer/artist give this diagram a refresh (my 2 cents).

Reply:
good call; well add that in when its complete.